src/Cache/Engine/DebugEngine.php

```diff
@@ -296,21 +291,21 @@ public function groups(): array
      * @param mixed $default The return value when the key does not exist.
      * @return mixed Config value being read.
      */
-    public function getConfig(?string $key = null, $default = null)
+    public function getConfig(?string $key = null, mixed $default = null): mixed
     {
         return $this->_engine->getConfig($key, $default);
     }
 
     /**
      * Sets the config.
      *
-     * @param string|array $key The key to set, or a complete array of configs.
+     * @param array|string $key The key to set, or a complete array of configs.
      * @param mixed|null $value The value to set.
      * @param bool $merge Whether to recursively merge or overwrite existing config, defaults to true.
      * @return $this
-     * @throws \Cake\Core\Exception\Exception When trying to set a key that is invalid.
+     * @throws \Cake\Core\Exception\CakeException When trying to set a key that is invalid.
      */
-    public function setConfig($key, $value = null, $merge = true)
+    public function setConfig(string|array $key, mixed $value = null, bool $merge = true)
```

Comment:
Unfortunatly, the sniff that converts annotations to type declarations seems to run first so we get out-of-order type declarations. I just searched and replaces the 2 most common: string|array and array|ArrayAccess.